bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/DoorbirdHandler.java

```diff
@@ -0,0 +1,622 @@
+        updateState(channelId, UnDefType.NULL);
+    }
+
+    // Get an array list of history images
+    private ArrayList<BufferedImage> getImages(String event) {
+        ArrayList<BufferedImage> images = new ArrayList<>();
+        int numberOfImages = config.montageNumImages;
+        String url = buildUrl("/bha-api/history.cgi", "?event=" + event + "&index=");
+        for (int imageNumber = 1; imageNumber <= numberOfImages; imageNumber++) {
+            String imageUrl = url + String.valueOf(imageNumber);
+            logger.debug("Downloading '{}' montage image {}", event, imageNumber);
+            DoorbirdImage historyImage = downloadImage(imageUrl);
+            if (historyImage != null) {
+                RawType image = historyImage.getImage();
+                if (image != null) {
+                    try {
+                        BufferedImage i = ImageIO.read(new ByteArrayInputStream(image.getBytes()));
+                        images.add(i);
+                    } catch (IOException e) {
+                        logger.debug("IOException creating BufferedImage from downloaded image: {}", e.getMessage());
+                    }
+                }
+            }
+        }
+        if (images.size() < numberOfImages) {
+            logger.debug("Some images could not be downloaded: wanted={}, actual={}", numberOfImages, images.size());
+        }
+        return images;
+    }
+
+    // Assemble the array of images into a single scaled image
+    private @Nullable State createMontage(ArrayList<BufferedImage> images) {
+        int height = (int) (images.get(0).getHeight() * (config.montageScaleFactor / 100.0));
+        int width = (int) (images.get(0).getWidth() * (config.montageScaleFactor / 100.0));
+        int widthTotal = width * images.size();
+        logger.debug("Dimensions of final montage image: w={}, h={}", widthTotal, height);
+
+        // Create concatenated image
+        int currentWidth = 0;
+        BufferedImage concatImage = new BufferedImage(widthTotal, height, BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2d = concatImage.createGraphics();
+        logger.debug("Concatenating images array into single image");
+        for (int j = 0; j < images.size(); j++) {
+            g2d.drawImage(images.get(j), currentWidth, 0, width, height, null);
+            currentWidth += width;
+        }
+        g2d.dispose();
+
+        // Convert image to a state
+        logger.debug("Rendering image to byte array and converting to RawType state");
+        byte[] imageBytes = convertImageToByteArray(concatImage);
+        State state = null;
+        if (imageBytes != null) {
+            state = new RawType(imageBytes, "image/png");
+        }
+        return state;
+    }
+
+    private byte @Nullable [] convertImageToByteArray(BufferedImage image) {
+        byte[] data = null;
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
```

Comment:
You can use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement here.

Reply:
Done